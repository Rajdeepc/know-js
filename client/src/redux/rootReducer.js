import { combineReducers } from "redux";
import AuthReducer from "../views/Auth/auth.reducer";
import NavbarReducer from '../components/Navbar/navbar.reducer'
import VideoPageReducer from '../views/VideoPage/video.reducer'
import HomePageReducer from '../views/Home/home.reducer'

const rootReducer =  combineReducers({
    AuthReducer,
    NavbarReducer,
    VideoPageReducer,
    HomePageReducer
})

export default rootReducer