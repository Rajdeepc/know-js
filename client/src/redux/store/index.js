import { createStore, applyMiddleware } from "redux"; // for creating store and applying middlerware like thunk
import thunk from "redux-thunk"; // for async actions
import rootReducer from "../rootReducer";
import { loadState, saveState } from "./sessionStorage";
import throttle from 'lodash/throttle';



const persistedState = loadState();

const store = createStore(
  rootReducer, 
  persistedState, 
  applyMiddleware(thunk));

store.subscribe(throttle(() => {
  saveState({
    LoginReducer: store.getState().LoginReducer
  });
},1000 ));


export default store;
