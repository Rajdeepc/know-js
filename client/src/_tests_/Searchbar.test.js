import React from "react";
import { render, cleanup } from "@testing-library/react";
import configureStore from "redux-mock-store";
import renderer from "react-test-renderer";
import SearchBarComponent from "../components/SearchBar/SearchBar";
import "@testing-library/jest-dom/extend-expect";
import { Provider } from "react-redux";
import { createStore } from "redux";
import rootReducer from "../redux/rootReducer";
import { BrowserRouter as Router } from "react-router-dom";

afterEach(cleanup);

const store = createStore(rootReducer);

describe("Connected Redux <SearchBarComponent component>", () => {
  const component = renderer.create(
    <Router>
      <Provider store={store}>
        <SearchBarComponent />
      </Provider>
    </Router>
  );
  test("renders SearchBarComponent Component", () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});
