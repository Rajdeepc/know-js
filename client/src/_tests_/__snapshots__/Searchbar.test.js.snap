// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Connected Redux <SearchBarComponent component> renders SearchBarComponent Component 1`] = `
<form
  className=""
  noValidate={true}
>
  <input
    className="form-control"
    onChange={[Function]}
    placeholder="Search here.."
    required={true}
    type="email"
  />
  <svg
    fill="none"
    height="1em"
    stroke="currentColor"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth="2"
    style={
      Object {
        "color": undefined,
      }
    }
    viewBox="0 0 24 24"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <circle
      cx="11"
      cy="11"
      r="8"
    />
    <line
      x1="21"
      x2="16.65"
      y1="21"
      y2="16.65"
    />
  </svg>
</form>
`;
